indexers/vector/FaissIndexer/__init__.py

```diff
@@ -76,16 +80,25 @@ def build_advanced_index(self, vecs: 'np.ndarray'):
             if train_data is None:
                 self.logger.warning('loading training data failed. some faiss indexes require previous training.')
             else:
-                self.train(index, train_data.astype(np.float32))
+                train_data = train_data.astype(np.float32)
+                if self.normalize:
+                    normalize_L2(train_data)
```

Comment:
are we sure normalize does the normalization in place?

Reply:
Yes, I read it and tested it.